### PR TITLE
feat: support skipping image creation

### PR DESCRIPTION
**Pull request by jkoelker**
Adds the config option `skip_create_image` to the builders to support
skipping image creation.

Fixes: #180

### CODE REVIEW

File: builder/azure/common/config.go
Comment (msdolbey) [inline builder/azure/common/config.go:16]: Should the default be first?

```suggestion
	// Defaults to `false`.
	// Useful for setting to `true` during a build test stage.
```

**Review by msdolbey — APPROVED** [review]
👍

File: builder/azure/common/config.go
Comment (jkoelker) [reply builder/azure/common/config.go:16]: It looks like in the rest of the documentation the defaults is the last thing mentioned. I think we should keep this new one consistent, but would be happy to reorg it if desired by the maintainers.

Comment (msdolbey) [reply builder/azure/common/config.go:16]: Oh, I just saw it different in some other spots - but def trust you on this one - we can resolve this.

**Comment by jkoelker** [comment]
Please let me know if this shared step approach is desirable. For CI purposes, I really only need something for the `azure-arm` builder, and if desired I can narrow the scope to only skipping the image capture for that builder.

**Comment by jkoelker** [comment]
I looked into the CI failures and it looks like they are existing in the repo due to commented out code. Please let me know if you want me to fix them. I took a stab at it briefly, but realized that the commented code and related issues with it may be part of a larger plan that I'm unaware of.

**Comment by nywilken** [comment]
Thanks for opening up this PR. We'll work on getting this reviewed as soon as possible to get you feedback on the changes.

**Comment by garimakhulbe02** [comment]
@jkoelker any update here? We also need this merge :)

**Comment by garimakhulbe02** [comment]
@nywilken any update? We need this merge too :)

**Comment by garimakhulbe02** [comment]
@nywilken Can you prioritize this PR? We have customers block on this.

File: builder/azure/common/config_test.go
Comment (JenGoldstrich) [inline builder/azure/common/config_test.go:13]: I think we should add a test case here where we pass in false for SkipCreateImage, but this is good for now.

**Review by JenGoldstrich — APPROVED** [review]
Hi @jkoelker and all,

I've spent a bit of time today getting up to speed on this PR, testing it for myself, and thinking about the change.

Overall it's a very good, kudos @jkoelker! I pointed out a place where we could add another unit test, and I think it would be good to include some example packer template files for this field, but I do not want to block merging of this PR on either of those issues.

I will look into back filling those two things soon, but would of course appreciate a PR adding them.

Merging.

**Comment by garimakhulbe02** [comment]
@jkoelker @JenGoldstrich Thank you!

**Pull request merged**
